crates/mun_compiler/src/driver.rs

```diff
@@ -104,14 +109,25 @@ impl Driver {
 
     /// Emits all diagnostic messages currently in the database; returns true if errors were
     /// emitted.
-    pub fn emit_diagnostics(&self, writer: &mut impl WriteColor) -> Result<bool, failure::Error> {
+    pub fn emit_diagnostics(
+        &self,
+        writer: &mut dyn std::io::Write,
+    ) -> Result<bool, failure::Error> {
         let mut has_errors = false;
+        let dlf = DisplayListFormatter::new(self.color_option.should_enable(), false);
         for file_id in self.db.source_root(WORKSPACE).files() {
             let diags = diagnostics(&self.db, file_id);
-            for diagnostic in diags.iter() {
-                diagnostic.emit(writer, &self.db, file_id)?;
-                if diagnostic.level == Level::Error {
-                    has_errors = true;
+            for diagnostic in diags {
+                let dl = DisplayList::from(diagnostic.clone());
+                writeln!(writer, "{}", dlf.format(&dl)).unwrap();
+                if let Some(annotation) = diagnostic.title {
+                    #[allow(clippy::single_match)]
+                    match annotation.annotation_type {
```

Comment:
Either handle all cases explicitly, or make this an `if let AnnotationType::Error = annotation.annotation_type`. This is preferable imo because now you are implicitly hiding changes in the implementation. E.g. if someone were to add `AnnotationType::Fatal`, we would probably want to set `has_errors = true`. if someone were to read this now, they'd think: "probably the previous author skipped all other cases on purpose". By being explicit about all cases, the compiler will throw an error if in the future - if a new `AnnotationType` is added - or if you use `if let` someone reading the source code will notice that `AnnotationType::Fatal` is not explicitly handled.